### PR TITLE
 [table service][client] Provide a simple version of storage clients

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

*Motivation*

The PR #1721 provides a routing service for proxying grpc requests
to the correct storage container. That means that the storage client
can be implemented using simple grpc calls since all containers can
serve same grpc requests.

*Changes*

Implement a simple version of storage clients by using server-side
routing to route grpc requests to the correct storage containers.

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [x] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [x] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [x] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [x] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [x] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [x] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java9]: skip build on java9. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

**Comment by sijie** [comment]
This PR is based on #1719 #1720 #1722 and gitsha 59dd8f3 is the change to review in this PR.

**Comment by sijie** [comment]
run pr validation

File: stream/server/src/main/java/org/apache/bookkeeper/stream/server/StorageServer.java [hunk 1]
```diff
@@ -249,6 +253,10 @@ public static LifecycleComponent buildStorageServer(CompositeConfiguration conf,
             dlConf,
             rootStatsLogger.scope("dlog"));
 
+        // client settings for the proxy channels
+        StorageClientSettings proxyClientSettings = StorageClientSettings.newBuilder()
+            .serviceUri("bk://localhost:" + grpcPort)
```
[the remark this answers is not in the document]
Comment (jiazhai) [reply stream/server/src/main/java/org/apache/bookkeeper/stream/server/StorageServer.java]: seems this changes brings in #1721

[the remark this answers is not in the document]
Comment (eolivelli) [reply stream/server/src/main/java/org/apache/bookkeeper/stream/server/StorageServer.java]: Sorry. I will port my comment to that commit

**Pull request merged**